### PR TITLE
Main 20 room management

### PR DESCRIPTION
**Pull request by kevo1ution**

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  } catch (err) {
+    if (err.response.status === StatusCodes.NOT_FOUND) {
+      return false;
+    }
+    throw err;
+  }
+};
+
+export const joinRoom = async (roomId: String) => {
+  const res = await axios.post(`/api/room/${roomId}/join`);
```
Comment (sarahannali) [inline src/models/room.ts:26]: how is this joining what info goes in here

File: src/models/room.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  return res.data.rooms;
+};
+
+export const getRoom = async (roomId: String) => {
+  const res = await axios.get(`/api/room/${roomId}`);
+  return res.data.room;
+};
+
+export const getRoomUsers = async (roomId: String) => {
+  const res = await axios.get(`/api/room/${roomId}/user`);
```
Comment (sarahannali) [inline src/models/room.ts:46]: should this route be /users

File: src/gameView/gamePlayer/GameInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            Create Room
+          </Button>
+          <UserContext.Consumer>
+            {({ user }) => (
+              <List>
+                <Typography variant="h4">
+                  {`${rooms.length} Active Rooms`}
+                </Typography>
+                {rooms.map((room: Room) => {
+                  const listContent = `Join Room: ${room.id.substr(room.id.length - 5)}`;
```
[the remark this answers is not in the document]
Comment (kevo1ution) [reply src/gameView/gamePlayer/GameInfo.tsx:74]: I wanted it to make it look cleaner and the room only shows the last 5 characters for the room id

File: src/gameView/gamePlayer/GameInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                  return (
+                    <ListItem
+                      key={room.id}
+                      button
+                      onClick={async (ev) => {
+                        if (!user) return;
```
[the remark this answers is not in the document]
Comment (kevo1ution) [reply src/gameView/gamePlayer/GameInfo.tsx:80]: user can be null. We want to do nothing in that situation

File: src/models/room.ts (hunk 1 above)
Comment (kevo1ution) [reply src/models/room.ts:26]: axios attaches the user information in authorization headers. The server will create the roomUser link

File: src/models/room.ts (hunk 2 above)
Comment (kevo1ution) [reply src/models/room.ts:46]: we used singular route nouns "/room", and I wanted to be consistent

**Review by sarahannali — APPROVED** [review]

**Pull request merged**